packages/back-end/src/util/mongo.util.ts

```diff
@@ -147,7 +146,11 @@ export function removeMongooseFields<T>(
   if (doc.toJSON) {
     doc = doc.toJSON({ flattenMaps: true });
   }
-  return omit(doc, ["_id", "__v"]) as T;
+
+  // Delete mongoose fields rather than using lodash.omit for perf reasons since this is called a lot
+  delete doc._id;
+  delete doc.__v;
```

Comment:
The usual concern with modifying the object in place is value sharing.

Are we sure that `doc` is never used concurrently in other places? For instance, can this mess up a concurrent `update` or `find` call that expects the `doc` object to also have those elements present?

For having been in such position before, tracking down these type of imperative issues is very hard to do so, usually, the performance gains have to be weighted against the risk of falling into such a situation.